dashboard/assets/packages/ui/src/components/Chart/Chart.hooks.ts

```diff
@@ -0,0 +1,20 @@
+import { useState } from "react"
+import uPlot, { type Options, type Series } from "uplot"
+
+import { mergeRightProps } from "utils"
+
+import { createOptions, CreateOptionsProps } from "./Chart.utils"
+
+const mergeRightShowProp = mergeRightProps(["show"])
+
+const mergeSeries = (plotSeries: Series[] = [], stateSeries: Series[] = []) => {
+  return plotSeries.map((series, i) => mergeRightShowProp(series, stateSeries[i]))
+}
+
+export const useOptions = ({ plot, theme, width }: CreateOptionsProps): Options => {
+  const [series, setSeries] = useState<Series[]>(plot.series)
+  const newPlot = { ...plot, series: mergeSeries(plot.series, series) }
```

Comment:
@szkiba this might be overkill, is it possible for plot.series to change over time?

Reply:
I have no idea (uPlot documentation is a little bit limited)